qiskit/extensions/standard/swap.py

```diff
@@ -55,8 +55,20 @@ def swap(self, ctl, tgt):
     if isinstance(ctl, QuantumRegister) and \
        isinstance(tgt, QuantumRegister) and len(ctl) == len(tgt):
         instructions = InstructionSet()
-        for i in range(ctl.size):
-            instructions.add(self.swap((ctl, i), (tgt, i)))
+        for j in range(ctl.size):
+            instructions.add(self.swap((ctl, j), (tgt, j)))
+        return instructions
+
+    if isinstance(ctl, QuantumRegister):
+        instructions = InstructionSet()
+        for j in range(ctl.size):
+            instructions.add(self.swap((ctl, j), tgt))
+        return instructions
+
+    if isinstance(tgt, QuantumRegister):
+        instructions = InstructionSet()
+        for j in range(tgt.size):
+            instructions.add(self.swap(ctl, (tgt, j)))
```

Comment:
The `swap` gate should not allow ctrl and tgt of different sizes I think.

I think swaps between 2 individual qubits or between 2 registers of the same size makes sense. But when the size of registers differ, an exception should be raised.

@awcross1 do you agree?

Reply:
@ajavadia Yes, I missed this. I agree that swap should not allow different sized registers as input. Good catch.

Reply:
Generally we should be careful when G_{i,j} and G_{i,k} do not commute. Does this happen anywhere else in the PR?

Reply:
fixed in 42f6b5a. The valid and invalid cases are listed in the tests. I'm not fully sure what you mean with  "G_{i,j} and G_{i,k} cases".

Reply:
Thanks @1ucian0 the issue doesn't occur anywhere else. I meant G_{i,j} to denote a two-qubit gate acting on qubits i and j.